module/state_synchronization/indexer/indexer_core.go

```diff
@@ -104,28 +103,23 @@ func (c *IndexerCore) IndexBlockData(data *execution_data.BlockExecutionDataEnti
 	})
 
 	g.Go(func() error {
-		updates := make([]*ledger.TrieUpdate, 0)
-
-		for _, chunk := range data.ChunkExecutionDatas {
-			if chunk.TrieUpdate != nil {
-				updates = append(updates, chunk.TrieUpdate)
-			}
-		}
-
 		// we are iterating all the registers and overwrite any existing register at the same path
 		// this will make sure if we have multiple register changes only the last change will get persisted
 		// if block has two chucks:
 		// first chunk updates: { X: 1, Y: 2 }
 		// second chunk updates: { X: 2 }
 		// then we should persist only {X: 2: Y: 2}
 		payloads := make(map[ledger.Path]*ledger.Payload)
-		for _, update := range updates {
-			for i, path := range update.Paths {
-				payloads[path] = update.Payloads[i]
+		for _, chunk := range data.ChunkExecutionDatas {
+			update := chunk.TrieUpdate
+			if update != nil && len(update.Paths) == len(update.Payloads) {
```

Comment:
We shouldn't silently ignore any updates, even if they are invalid. Otherwise, the register db may no longer match ENs. This should either return an error or be unhandled causing a panic.

Realistically, this should never happen since the execution data has already been verified on VNs, and the hash checked on ANs after download